cypress/e2e/session-recording.cy.ts

```diff
@@ -362,6 +362,33 @@ describe('Session recording', () => {
                     })
                 })
         })
+
+        it('starts a new recording after calling reset', () => {
+            cy.phCaptures({ full: true }).then((captures) => {
+                // should be a pageview at the beginning
+                expect(captures.map((c) => c.event)).to.deep.equal(['$pageview'])
+            })
+            cy.resetPhCaptures()
+
+            let startingSessionId: string | null = null
+            cy.posthog().then((ph) => {
+                startingSessionId = ph.get_session_id()
+            })
+
+            ensureActivitySendsSnapshots()
+
+            cy.posthog().then((ph) => {
+                ph.reset()
+            })
+
```

Comment:
I couldn't call `ensureActivitySendsSnapshots()` again here because two custom events were missing: <img width="968" alt="Screenshot 2024-04-12 at 12 46 32" src="https://github.com/PostHog/posthog-js/assets/6685876/0fcb5bb0-7ca3-4fc2-b073-2b6614c3a8df">

I wasn't sure where these were coming from but assumed they may be fired after RRWeb initialises and hence won't fire a second time

Reply:
Let's alter that `ensure` or duplicate it so we can test for _any_ snapshots maybe?